app/src/main/java/com/example/recipeopedia/activities/RecipeDetailsActivity.java

```diff
@@ -3,38 +3,90 @@
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.databinding.DataBindingUtil;
 
-import android.content.Context;
+import android.content.Intent;
 import android.os.Bundle;
 import android.util.Log;
-import android.widget.ImageView;
-import android.widget.TextView;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
 
-import com.bumptech.glide.Glide;
 import com.example.recipeopedia.R;
 import com.example.recipeopedia.databinding.ActivityRecipeDetailsBinding;
+import com.example.recipeopedia.models.FavoriteRecipe;
 import com.example.recipeopedia.models.Recipe;
+import com.parse.ParseException;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+import com.parse.SaveCallback;
 
 import org.parceler.Parcels;
 
 public class RecipeDetailsActivity extends AppCompatActivity {
     public static final String TAG = "RecipeDetailsActivity";
     private Recipe recipe;
-    private TextView tvRecipeName, tvIngredients, tvInstructions;
-    private ImageView ivRecipeImage;
-
-    // TextView nutrition, mealType, cuisineType, dishType, dietLabels, healthLabels;
+    private Button btnSave, btnReview;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_recipe_details);
 
         recipe = Parcels.unwrap(getIntent().getParcelableExtra(Recipe.class.getSimpleName()));
-        Log.d("RecipeDetailsActivity", String.format("Showing details for '%s'", recipe.getRecipeName()));
-
         ActivityRecipeDetailsBinding binding =
                 DataBindingUtil.setContentView(this, R.layout.activity_recipe_details);
-        binding.setImageUrl(recipe.getImage());
         binding.setRecipe(recipe);
+
+        btnSave = findViewById(R.id.btnSave);
+        btnSave.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                try {
+                    saveRecipe(recipe.getRecipeName(), ParseUser.getCurrentUser(), recipe.getImage(),
+                            recipe.getIngredients(), recipe.getInstructions(), recipe.getHealthLabels());
```

Comment:
nit: readability suggestion
```suggestion
                    saveRecipe(
                        recipe.getRecipeName(), 
                        ParseUser.getCurrentUser(), 
                        recipe.getImage(),
                        recipe.getIngredients(), 
                        recipe.getInstructions(), 
                        recipe.getHealthLabels());
```